### PR TITLE
SSSDUtils: config file ownership should match service user

### PR DESCRIPTION
**Pull request by alexey-tikhonov**

### CODE REVIEW

File: sssd_test_framework/utils/sssd.py [hunk 1]
```diff
@@ -406,7 +407,12 @@ def config_apply(self, check_config: bool = True, debug_level: str | None = "0xf
         :type debug_level:  str | None, optional
         """
         cfg = self.__set_debug_level(debug_level)
-        service_user = self.config.get("sssd", "user", fallback="root")
+        if self.fs.exists("/etc/systemd/system/multi-user.target.wants/sssd.service"):
+            service_file = self.fs.read("/etc/systemd/system/multi-user.target.wants/sssd.service")
```
Comment (pbrezina) [inline sssd_test_framework/utils/sssd.py:411]: `systemctl cat sssd.service` would be better, its just one call and it also considers overwrites

Comment (alexey-tikhonov) [reply sssd_test_framework/utils/sssd.py:411]: This should be done by extension of https://pytest-mh.readthedocs.io/en/latest/api/pytest_mh.utils.services.html#pytest_mh.utils.services.SystemdServices ?

Comment (pbrezina) [reply sssd_test_framework/utils/sssd.py:411]: Not necessarily, but it would be nicer.

Comment (alexey-tikhonov) [reply sssd_test_framework/utils/sssd.py:411]: https://github.com/next-actions/pytest-mh/pull/40

Comment (pbrezina) [reply sssd_test_framework/utils/sssd.py:411]: Merged.

**Comment by alexey-tikhonov** [comment]
This doesn't break current sssd-2-10 branch - see CI results for https://github.com/SSSD/sssd/pull/7178 (centos-9)

For example following tests still run fine both with 'root' and 'sssd' 'sssd.conf::user' value.
```
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[root] (ipa) PASSED [ 24%]
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[root] (ldap) PASSED [ 24%]
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[root] (samba) PASSED [ 24%]
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[sssd] (ipa) PASSED [ 25%]
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[sssd] (ldap) PASSED [ 25%]
tests/test_identity.py::test_identity__lookup_group_membership_by_username_with_id[sssd] (samba) PASSED [ 25%]
```

But this is important for patches that make service user configurable via sssd.service::User

**Review by pbrezina — APPROVED** [review]

**Pull request merged**